book_config.py

```diff
@@ -130,75 +111,43 @@ def __init__(self, db, ids, parent):
     @property
     def book(self):
         return self._book_settings[self._index]
-    
-    def edit_asin(self, val):
-        self.book.asin = val
 
-    def edit_shelfari_url(self, val):
-        http_string = 'http://www.'
+    def edit_goodreads_url(self, val):
+        goodreads_string = 'https://www.goodreads.com/'
         index = 0
-        if val[:len(http_string)] != http_string:
+        if val[:len(goodreads_string)] != goodreads_string:
             for i, letter in enumerate(val):
-                if i < len(http_string):
-                    if letter == http_string[i]:
+                if i < len(goodreads_string):
+                    if letter == goodreads_string[i]:
                             index += 1
                     else:
                         break
                 else:
                     break
-            self.shelfari_url_edit.setText(http_string + val[index:])
-
-        self.book.shelfari_url = val
-
-    def search_for_asin(self):
-        asin = None
-        asin = self.book.get_asin()
-        if not asin:
-            if self.book.prefs['asin'] != '':
-                self.status.setText('ASIN not found. Using original asin.')
-                self.asin_edit.setText(self.book.prefs['asin'])
-            else:
-                self.status.setText('ASIN not found.')
-                self.asin_edit.setText('')
-        else:
-            self.status.setText('ASIN found.')
-            self.book.asin = asin
-            self.asin_edit.setText(asin)
+            self.goodreads_url_edit.setText(goodreads_string + val[index:])
 
-    def search_for_shelfari_url(self):
+        self.book.goodreads_url = val
+
+    def search_for_goodreads_url(self):
         url = None
-        if self.asin_edit.text() != '' and self.asin_edit.text() != 'ASIN not found':
-            url = self.book.search_shelfari(self.asin_edit.text())
-        if not url:
-            if self.book._prefs['asin'] != '':
-                url = self.book.search_shelfari(self.book._prefs['asin'])
-        if not url:
-            if self.book.title != 'Unknown' and self.book.author != 'Unknown':
-                url = self.book.search_shelfari(self.book.title_and_author)
+        if self.book.title != 'Unknown' and self.book.author != 'Unknown':
+            url = self.book.search_goodreads(self.book.title_and_author)
```

Comment:
As commented elsewhere: for me searching by title (ie. entering title, and selecting the "title" radio button) gives good results, better than doing a search by title+author using "all" radio button.